### PR TITLE
Rectangle doc tweaks

### LINKED ISSUE #1368: unnest() doc tweaks
* Pull `hoist()` out into it's own page
* Pull `unnest_auto()` out into it's own page, and don't recommend it anywhere.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1368

### CODE REVIEW

**Comment by hadley** [comment]
Once this is merged, I'd also split up into more files; `rectangle.R` is way too big.

File: R/rectangle.R [hunk 1]
```diff
@@ -1,56 +1,13 @@
-#' Rectangle a nested list into a tidy tibble
+#' Hoist values out of list-columns
 #'
 #' @description
-#' `hoist()`, `unnest_longer()`, and `unnest_wider()` provide tools for
-#' rectangling, collapsing deeply nested lists into regular columns.
 #' `hoist()` allows you to selectively pull components of a list-column out
 #' in to their own top-level columns, using the same syntax as [purrr::pluck()].
```
Comment (DavisVaughan) [inline R/rectangle.R:5]:
```suggestion
#' `hoist()` allows you to selectively pull components of a list-column
#' into their own top-level columns, using the same syntax as [purrr::pluck()].
```

File: R/rectangle.R [hunk 2]
```diff
@@ -107,77 +64,14 @@
...
-#'
-#' unnest_longer(wide, c(x, y))
-#'
-#' # To instead enforce strict vctrs typing rules, use `strict`
-#' wide_strict <- unnest_wider(df, json, strict = TRUE)
-#' wide_strict
-#'
-#' try(unnest_longer(wide_strict, c(x, y)))
 #' @export hoist
+#' @family rectangling
```
Comment (DavisVaughan) [inline R/rectangle.R:74]: Screen Shot 2022-06-10 at 11 15 31 AM

FWIW this is what I see from using the family tag. I wish it showed `unnest()` but I guess that is ok

File: R/rectangle.R [hunk 3]
```diff
@@ -722,8 +732,25 @@ elt_to_wide <- function(x, name, strict, names_sep) {
...
+#'
+#' It's handy for very rapid interactive exploration but I don't recommend
+#' using it in scripts, because it will succeed even if the underlying data
+#' radically changes.
+#'
+#' @inheritParams unnest_longer
 #' @export
-#' @rdname hoist
+#' @param col List-column to extract components from.
 unnest_auto <- function(data, col) {
```
Comment (DavisVaughan) [inline R/rectangle.R:754]: How aggressively do you want to discourage this? We could `@keyword internal` it so it doesn't show up in the pkgdown docs?

File: R/rectangle.R [hunk 4]
```diff
@@ -722,8 +732,25 @@ elt_to_wide <- function(x, name, strict, names_sep) {
...
+#'
+#' * If all elements are unnamed, it uses
+#'   `unnest_longer(indices_include = FALSE)`.
+#' * If all elements are named, and there's at least one name in
+#'   common across all components, it uses `unnest_wider()`.
+#' * Otherwise, it falls back to `unnest_longer(indices_include = TRUE)`.
+#'
+#' It's handy for very rapid interactive exploration but I don't recommend
+#' using it in scripts, because it will succeed even if the underlying data
+#' radically changes.
```
Comment (DavisVaughan) [inline R/rectangle.R:749]: I think you need to redocument again because this section didn't show up for me in the doc page

**Review by DavisVaughan — APPROVED** [review]

File: R/rectangle.R (hunk 2 above)
Comment (hadley) [reply R/rectangle.R:74]: Yeah, maybe `nest() `and `unnest()` should be separate pages? I'll take a look

File: R/rectangle.R (hunk 3 above)
Comment (hadley) [reply R/rectangle.R:754]: Yeah, that's a good idea.

**Pull request merged**